### ISSUE TITLE
probably wrong answer from active_set

### ISSUE THREAD

**Issue by JianqiangDing** [libigl/libigl#1952]
#### Describe the bug

Hi, it seems that the active set method does not return the right answer for the following simple qp problem:

Here is the python script for your reference:

```python
    import numpy as np
    import scipy.sparse as sp
    import igl

    """
    min x.T * P * x + q.T * x
    s.t. G @ x <= h
         A @ x == b
    """

    m = np.array([[1., 2., 0.], [-8., 3., 2.], [0., 1., 1.]], dtype=float)
    p = np.matmul(m.T, m)
    q = np.matmul(np.array([3., 2., 3.], dtype=float), m).reshape((3,))
    g = np.array([[1., 2., 1.], [2., 0., 1.], [-1., 2., -1.]], dtype=float)
    h = np.array([3., 2., -2.], dtype=float).reshape((3,))
    a = np.array([1., 1., 1.], dtype=float)
    b = np.array([1.], dtype=float)
    lx = np.array([], dtype=float)
    ux = np.array([], dtype=float)
    z_b = np.array([], dtype=int)
    z_bc = np.array([], dtype=float)
    inr_p = sp.csr_matrix(p)
    inr_g = sp.csr_matrix(g)
    inr_a = sp.csr_matrix(a)
    state, x = igl.active_set(2 * inr_p, q, z_b, z_bc, inr_a, b, inr_g, h, lx, ux)
```
results:  x=[ 0.20879121 -1.79120879  2.58241758] and state=0 (SOLVER_STATUS_CONVERGED = 0 indicates the algorithm converaged as doc said)

while MATLAB returns different x as x = [0.2418   -0.7582    1.5165](and [cvxpy](https://www.cvxpy.org) results same answer as MATLAB), here is the MATLAB script:

```matalab
clc
clear
p=[
65.0  -22.0  -16.0  ;
-22.0  14.0  7.0  ;
-16.0  7.0  5.0  ;
];
q=[
-13.0  15.0  7.0  ;
];
g=[
1.0  2.0  1.0  ;
2.0  0.0  1.0  ;
-1.0  2.0  -1.0  ;
];
h=[
3.0  2.0  -2.0  ;
];
a=[
1.0  1.0  1.0  ;
];
b=1.0;

options = optimoptions('quadprog','Display','iter');
[x,fval,exitflag,output,lambda] = quadprog(2*p,q,g,h,a,b,[],[],[],options);
x
```
log from MATLAB
```
 Iter            Fval  Primal Infeas    Dual Infeas  Complementarity  
    0   -1.379495e-01   1.041584e+00   3.210657e+00     1.605329e+00  
    1   -2.892374e-01   9.629171e-03   2.968169e-02     4.274688e-02  
    2   -3.150369e-01   8.334087e-06   2.568963e-05     3.842655e-03  
    3   -3.186236e-01   4.167044e-09   1.284481e-08     5.776800e-05  
    4   -3.186813e-01   2.083667e-12   1.093703e-11     2.923162e-08  
    5   -3.186813e-01   2.442491e-15   8.881784e-15     1.461580e-11  

Minimum found that satisfies the constraints.

Optimization completed because the objective function is non-decreasing in 
feasible directions, to within the value of the optimality tolerance,
and constraints are satisfied to within the value of the constraint tolerance.

<stopping criteria details>

x =

    0.2418
   -0.7582
    1.5165

```
Is this a bug or [issue about linear independent of Aeq](https://github.com/libigl/libigl/blob/79dc4f683877a8bb4eaf63620437a5454f6c47c1/include/igl/active_set.h) (even though only one row here) or something else? 

#### Platform

- [ ] Windows
- [ ] macOS
- [x] Linux

**Comment by alecjacobson** [comment]
Looks like the linear *inequality* constraints are being violated. I seem to remember that those are not as well tested as the bound constraints. I also seem to remember that they're implemented in a way that is pretty expensive. Given that significantly better solvers now exist for big sparse QPs (e.g., nasoq https://nasoq.github.io/), I doubt that this will get fixed in igl. Perhaps we should add a quick check whether the solution satisfies the constraints and return a more appropriate status.

**Comment by ladzin** [comment]
I've also ran into active_set issues. Returned SOLVER_STATUS_CONVERGED, inequality constraints are satisfied, but the solution is very suboptimal. OSQP worked fine. Maybe tag the routine as "experimental" or some other warning like that?

**Comment by alecjacobson** [comment]
"Fixed" by 7c58cb041deb5ff5693fc63a7fe0c1241b0729a8
